tdms/src/argument_parser.cpp

```diff
@@ -0,0 +1,120 @@
+#include "stdexcept"
+#include "argument_parser.h"
+
+using namespace std;
+
+
+ArgumentParser::ArgumentParser() = default;
+
+ArgumentNamespace ArgumentParser::parse_args(int nargs, char *argv[]) {
+
+  auto args = ArgumentNamespace(nargs, argv);
+
+  if (args.have_flag("-h")){
+    print_help_message();
+    exit(0);
+  }
+
+  if( (args.num_non_flag != 2) &&  (args.num_non_flag != 3) ){
+    fprintf(stderr,"incorrect number of arguments\n");
+    exit(-1);
+  }
+
+  if( args.num_non_flag == 2){
+    fprintf(stdout,"infile:[%s], outfile:[%s], m=%d\n",
+            args.input_filename(),
+            args.output_filename(),
+            args.have_flag("-m"));
+  }
+  else{
+    fprintf(stdout,"infile:[%s], gridfile:[%s], outfile:[%s], m=%d\n",
+            args.input_filename(),
+            args.grid_filename(),
+            args.output_filename(),
+            args.have_flag("-m"));
+  }
+
+  return args;
+}
+
+void ArgumentParser::print_help_message(){
+  fprintf(stdout,"Usage:\n"
+                 "openandorder [options] infile outfile\n"
+                 "openandorder [options] infile gridfile outfile\n"
```

Comment:
Ideally (for simplicity) this would be
```suggestion
                 "openandorder [options] infile outfile gridfile\n"
```
to match the two-arg case. But I guess there's a good reason (backwards compat?) to keep it this way round?

Reply:
I fear we need some backwards compat, not to break _all_ of Peter's scripts